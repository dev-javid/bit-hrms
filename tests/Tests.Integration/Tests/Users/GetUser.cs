namespace Tests.Integration.Tests.Users
{
    public class GetUser(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/users";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Users/GetUser.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
