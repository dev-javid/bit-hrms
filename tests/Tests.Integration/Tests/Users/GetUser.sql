INSERT INTO public.users VALUES 
(101, 'super-admin1@example.com',   'SUPER-ADMIN1@EXAMPLE.COM',      'super-admin1@example.com',      'SUPER-ADMIN1@EXAMPLE.COM',      true, 'AQAAAAIAAYagAAAAEIDoeTRLrGQvn8FRF2EXGdQ5GtreAkKzZcbl1BTD6xNNzUXrGkn6BCPp05RxjIH2lQ==', '2XTX6Z5RVMLHRXNWIMW3UJ2USO524VVU', '283d4f58-3a75-47ad-bf9c-de04c9e9a80a', '7876543210', false, false, NULL, true, 0, 0, '2024-05-03 03:51:44.770234-07', NULL, NULL),
(201, 'company-admin1@example.com', 'COMPANY-ADMIN1@EXAMPLE.COM',    'company-admin1@example.com',    'COMPANY-ADMIN1@EXAMPLE.COM',    true, 'AQAAAAIAAYagAAAAEIDoeTRLrGQvn8FRF2EXGdQ5GtreAkKzZcbl1BTD6xNNzUXrGkn6BCPp05RxjIH2lQ==', '2XTX6Z5RVMLHRXNWIMW3UJ2USO524VVU', '283d4f58-3a75-47ad-bf9c-de04c9e9a80a', '7876543211', false, false, NULL, true, 0, 0, '2024-05-03 03:51:44.770234-07', NULL, NULL),
(301, 'employee1@example.com',      'EMPLOYEE1@EXAMPLE.COM',         'employee1@example.com',         'EMPLOYEE1@EXAMPLE.COM',         true, 'AQAAAAIAAYagAAAAEIDoeTRLrGQvn8FRF2EXGdQ5GtreAkKzZcbl1BTD6xNNzUXrGkn6BCPp05RxjIH2lQ==', '2XTX6Z5RVMLHRXNWIMW3UJ2USO524VVU', '283d4f58-3a75-47ad-bf9c-de04c9e9a80a', '7876543212', false, false, NULL, true, 0, 0, '2024-05-03 03:51:44.770234-07', NULL, NULL);