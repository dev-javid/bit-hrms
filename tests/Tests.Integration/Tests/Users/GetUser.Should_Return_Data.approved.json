{
    "page": 1,
    "limit": 10,
    "total": 6,
    "items": [
        {
            "userId": 100,
            "email": "super-admin@example.com",
            "phoneNumber": "9876543210",
            "claims": []
        },
        {
            "userId": 101,
            "email": "super-admin1@example.com",
            "phoneNumber": "7876543210",
            "claims": []
        },
        {
            "userId": 200,
            "email": "company-admin@example.com",
            "phoneNumber": "9876543211",
            "claims": []
        },
        {
            "userId": 201,
            "email": "company-admin1@example.com",
            "phoneNumber": "7876543211",
            "claims": []
        },
        {
            "userId": 300,
            "email": "employee@example.com",
            "phoneNumber": "9876543212",
            "claims": []
        },
        {
            "userId": 301,
            "email": "employee1@example.com",
            "phoneNumber": "7876543212",
            "claims": []
        }
    ]
}